src/repo.rs

```diff
@@ -161,6 +161,74 @@ impl Repo {
         self.update_hash_on_dht(&hash).await?;
         Ok(hash)
     }
+
+    // Method to get or create a collection associated with the repo
+    pub async fn get_or_create_collection(&self) -> Result<Hash> {
+        self.check_write_permissions()?;
+        let collection_name = self.get_name().await?;
+
+        match self.iroh_blobs.collection_hash(&collection_name).await {
+            Ok(collection_hash) => Ok(collection_hash),
+            Err(_) => self.iroh_blobs.create_collection(&collection_name).await,
+        }
+    }
+
+    // Method to add a file to the collection identified by repo ID
+    pub async fn set_file_in_repo_collection(&self, file_name: &str, file_hash: Hash) -> Result<Hash> {
+        self.check_write_permissions()?;
+        let collection_name = self.get_name().await?;
+
+        self.iroh_blobs.set_file(&collection_name, file_name, &file_hash).await
+    }
+
+    // Method to retrieve a file's hash from the collection
+    pub async fn get_file_from_repo_collection(&self, file_name: &str) -> Result<Hash> {
+        let collection_name = self.get_name().await?;
+
+        self.iroh_blobs.get_file(&collection_name, file_name).await
+    }
+
+    // Method to list all files in the collection
+    pub async fn list_files_in_repo_collection(&self) -> Result<Vec<String>> {
+        let collection_name = self.get_name().await?;
+
+        self.iroh_blobs.list_files(&collection_name).await
+    }
+
+    // Method to delete a file from the collection
+    pub async fn delete_file_from_repo_collection(&self, file_name: &str) -> Result<Hash> {
+        self.check_write_permissions()?;
+        let collection_name = self.get_name().await?;
+
+        self.iroh_blobs.delete_file(&collection_name, file_name).await
+    }
+
+    // Method to get the collection's hash
+    pub async fn get_collection_hash(&self) -> Result<Hash> {
```

Comment:
I think we can keep this private just for setting the hash when we first boot up the repo if can_write

```suggestion
   async fn get_collection_hash(&self) -> Result<Hash> {
```